IMOS_ATF-ACOUSTIC/index_tmp.sql

```diff
@@ -0,0 +1,47 @@
+-- insert what has been harvested into the database with index_tmp
+-- creates if needed the table detection_file_index
+-- then creates the views to be used for WMS
+
+CREATE TABLE IF NOT EXISTS detection_file_index
+(
+    -- id bigserial PRIMARY KEY,
+    tag_deployment_id TEXT PRIMARY KEY, --UNIQUE NOT NULL,
+    transmitter_id TEXT,
+    species TEXT,
+    tag_project TEXT,
+    time_coverage_start timestamp with time zone,
+    time_coverage_end timestamp with time zone,
+    geom geometry(Geometry,4326),
+    first_indexed timestamp with time zone DEFAULT Now(),
+    last_indexed timestamp with time zone DEFAULT Now(),
+    url TEXT,
+    size integer
+);
+
+-- -- create the geometry column for a polygon
+-- --SET search_path = harvest, public;
+-- SELECT AddGeometryColumn('animal_tracking_acoustic_qc','index_tmp','geom',4326,'POLYGON',2);
+-- UPDATE animal_tracking_acoustic_qc.index_tmp
+--        SET geom = ST_GeomFromText('POLYGON(('|| min_lon::text ||' '|| min_lat::text ||','|| min_lon::text ||' '|| max_lat::text ||',
+-- '|| max_lon::text ||' '|| max_lat::text ||','|| max_lon::text ||' '|| min_lat::text ||', '|| min_lon::text ||' '|| min_lat::text ||'))', 4326);
+
+-- read geom from dataframe - trajectory 
+SELECT AddGeometryColumn('animal_tracking_acoustic_qc','index_tmp','geom',4326,'LINESTRING',2);
+UPDATE index_tmp
+       SET geom = ST_GeomFromText(geom_df, 4326);
+
+-- UPSERT
+-- insert newly harvested record from index_tmp to the main table
+-- 1. if tag_deployment_id does not exist --> new entry, insert everything
+-- 2. if tag_deployment_id exists --> update record where needed
+insert into detection_file_index (tag_deployment_id, transmitter_id, species, tag_project,
+                                  time_coverage_start, time_coverage_end, url, size, geom)
+SELECT tag_deployment_id, transmitter_id, species, tag_project, time_coverage_start, time_coverage_end,
+       url, size, geom
+FROM index_tmp
+ON CONFLICT (tag_deployment_id)
+DO
+    UPDATE SET time_coverage_start = excluded.time_coverage_start, time_coverage_end = excluded.time_coverage_end,
+               last_indexed = now()::timestamp, size = excluded.size,
+               geom = excluded.geom
```

Comment:
Is it safe to assume that `transmitter_id`, `species`, `tag_project` and `url` aren't going to change?

Reply:
Yes and in a way they're helping creating the `tag_deployment_id` which is unique. Each csv file will be for one tag/one animal/one project

Reply:
:+1: